src/osd/osd_types.h

```diff
@@ -523,7 +530,8 @@ class coll_t {
   spg_t pgid;
   uint64_t removal_seq;  // note: deprecated, not encoded
 
-  string _str;  // cached string
+  char _str_buff[spg_t::calc_name_buf_size];
+  char *_str = NULL;
```

Comment:
I don't think you need to initialize _str, calc_str() will always take care of that.
But even if you need to you can always initialize it in the constructor initialization list.
With _str code will look cleaner than with _str_offset.